src/media/local-stream.ts

```diff
@@ -78,44 +68,53 @@ abstract class _LocalStream extends Stream {
   setMuted(isMuted: boolean): void {
     if (this.inputTrack.enabled === isMuted) {
       this.inputTrack.enabled = !isMuted;
+      // setting `enabled` will not automatically emit MuteStateChange, so we emit it here
       this[StreamEventNames.MuteStateChange].emit(isMuted);
     }
   }
 
   /**
-   * Get the settings of the output track on this stream.
-   *
-   * @returns The settings of the track.
+   * @inheritdoc
    */
   getSettings(): MediaTrackSettings {
-    return this.outputTrack.getSettings();
+    return this.inputTrack.getSettings();
   }
 
   /**
-   * Get the label of the output track on this stream.
+   * Get the label of the input track on this stream.
    *
    * @returns The label of the track.
    */
   get label(): string {
-    return this.outputTrack.label;
+    return this.inputTrack.label;
   }
 
   /**
-   * Replace the existing track on the output stream for a new track.
+   * Change the track of the output stream to a different track. If the new track is the input
+   * track, then set the output stream to be the input stream. If the input and output tracks are
+   * currently the same, then set the output stream to a new stream with the new track. Otherwise,
+   * remove the old track from the output stream and add the new one.
    *
-   * @param newTrack - The track to add to the stream.
+   * @param newTrack - The track to be used in the output stream.
    */
-  private replaceTrack(newTrack: MediaStreamTrack): void {
-    this.outputStream.removeTrack(this.outputTrack);
-    this.outputStream.addTrack(newTrack);
+  private changeOutputTrack(newTrack: MediaStreamTrack): void {
+    if (this.inputTrack.id === newTrack.id) {
+      this._outputStream = this.inputStream;
+    } else if (this.inputTrack.id === this.outputTrack.id) {
+      this._outputStream = new MediaStream([newTrack]);
```

Comment:
We have to preserve the output stream, since that's what callers will have a handle to.  So we actually need to create a new stream that'll be used as the input stream, and move the current input/output track to that, and then add the new track to the existing outputstream.  (Tweak the comment as well)

Reply:
I'd think enabling background blur or something should show that this doesn't work: the local rendering of the video wouldn't show the blur since the outputstream changes?